bcipy/task/paradigm/rsvp/copy_phrase.py

```diff
@@ -715,6 +722,55 @@ def btn_press_count(self) -> int:
             activations = [inq for inq in inquiries if not inq.eeg_evidence]
         return len(activations)
 
+    def switch_response_time(self) -> Optional[float]:
+        """Computes the average switch response in seconds."""
+
+        # Remove consecutive items with the same type; we are only interested
+        # in PREVIEW followed by a KEY_PRESS.
+        triggers = destutter(self.switch_triggers(), key=lambda trg: trg.type)
+        pairs = list(zip(triggers[::2], triggers[1::2]))
+
+        # Confirm that the data is structured as expected.
+        for preview, keypress in pairs:
+            valid = preview.type == TriggerType.PREVIEW and keypress.type == TriggerType.KEY_PRESS
+            if not valid:
+                self.logger.info("Could not compute switch_response_time")
+                return None
+
+        response_times = [
+            keypress.time - preview.time for preview, keypress in pairs
+        ]
+        count = len(response_times)
+        return sum(response_times) / count if count > 0 else None
+
+    def switch_triggers(self) -> List[Trigger]:
+        """Returns a list of switch-related triggers"""
+        if not self.trigger_path:
+            return []
+        triggers, _offset = TriggerHandler.read_text_file(self.trigger_path)
+        return [
+            trg for trg in triggers
+            if trg.type in [TriggerType.PREVIEW, TriggerType.KEY_PRESS]
+        ]
+
+
+def destutter(items: List, key: Callable = lambda x: x) -> List:
```

Comment:
This should live somewhere else... helpers? I can suggest something if one doesn't come to mind.

Reply:
This function seemed a bit too generic for any of our existing helpers so I went ahead and created a new one. If you would rather see it elsewhere let me know.